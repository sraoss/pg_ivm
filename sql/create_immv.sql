CREATE TABLE t (i int PRIMARY KEY);
INSERT INTO t SELECT generate_series(1, 100);

SELECT create_immv('mv', 'SELECT * FROM t');
SELECT create_immv(' mv2 ( x  ) ', 'SELECT * FROM t WHERE i%2 = 0');

SELECT create_immv('mv3', 'WITH d AS (DELETE FROM t RETURNING NULL) SELECT * FROM t');

SELECT immvrelid FROM pg_ivm_immv ORDER BY 1;

DROP TABLE t;

DROP TABLE mv;
SELECT immvrelid FROM pg_ivm_immv ORDER BY 1;

DROP TABLE mv2;
SELECT immvrelid FROM pg_ivm_immv ORDER BY 1;

DROP TABLE t;
