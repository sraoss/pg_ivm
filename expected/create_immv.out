CREATE TABLE t (i int PRIMARY KEY);
INSERT INTO t SELECT generate_series(1, 100);
SELECT create_immv('mv', 'SELECT * FROM t');
NOTICE:  created index "mv_index" on immv "mv"
 create_immv 
-------------
         100
(1 row)

SELECT create_immv(' mv2 ( x  ) ', 'SELECT * FROM t WHERE i%2 = 0');
NOTICE:  created index "mv2_index" on immv "mv2"
 create_immv 
-------------
          50
(1 row)

SELECT create_immv('mv3', 'WITH d AS (DELETE FROM t RETURNING NULL) SELECT * FROM t');
ERROR:  materialized views must not use data-modifying statements in WITH
SELECT immvrelid FROM pg_ivm_immv ORDER BY 1;
 immvrelid 
-----------
 mv
 mv2
(2 rows)

DROP TABLE t;
ERROR:  cannot drop table t because other objects depend on it
DETAIL:  table mv depends on table t
table mv2 depends on table t
HINT:  Use DROP ... CASCADE to drop the dependent objects too.
DROP TABLE mv;
SELECT immvrelid FROM pg_ivm_immv ORDER BY 1;
 immvrelid 
-----------
 mv2
(1 row)

DROP TABLE mv2;
SELECT immvrelid FROM pg_ivm_immv ORDER BY 1;
 immvrelid 
-----------
(0 rows)

DROP TABLE t;
